spec/test_tag_spec.rb

```diff
@@ -0,0 +1,74 @@
+require 'rspec'
+require_relative '../test_tag_generator'
+
+describe 'TestTagGenerator' do
+  before(:each) do
+    @test_tag_generator = TestTagGenerator.new
+  end
+
+  describe 'check_for_tag?' do
+    before(:each) do
+      @check_tag_type = 'dev'
+    end
+
+    context 'when there are no develop tags on the commit but there are other tags' do
```

Comment:
Ah yes, I forgot to do this. I think it would be better to catch this before a second dev tag is applied to a commit. Unless you want to be able to apply multiple tag types to a single commit?